base/strings/search.jl

```diff
@@ -334,18 +335,16 @@ findlast(ch::AbstractChar, string::AbstractString) = findlast(==(ch), string)
 
 # AbstractString implementation of the generic findprev interface
 function findprev(testf::Function, s::AbstractString, i::Integer)
-    if i < 1
-        return i == 0 ? nothing : throw(BoundsError(s, i))
-    end
-    n = ncodeunits(s)
-    if i > n
-        return i == n+1 ? nothing : throw(BoundsError(s, i))
+    i = Int(i)
+    z = ncodeunits(s) + 1
+    0 ≤ i ≤ z || throw(BoundsError(s, i))
+    i == z && return nothing
```

Comment:
This matches the old behavior, but it mystifies me.   Was it intentional, or a bug?

Reply:
(Note that `findnext` and `findprev` for `Array` both throw a `BoundsError` if the starting index is invalid, so the string functions are inconsistent.)

Reply:
I remember reading something about allowing indexing strings one byte after the end or something but don't recall the details. @bkamins or @StefanKarpinski might know.

Reply:
https://github.com/JuliaLang/julia/pull/24999#issuecomment-350492880

Reply:
Okay.